src/Data/Vec/Base.agda

```diff
@@ -345,12 +345,12 @@ initLast {n = suc n} (x ∷ xs) with initLast xs
 ... | (ys , y , refl) = (x ∷ ys , y , refl)
 
 init : Vec A (1 + n) → Vec A n
-init xs with initLast xs
-... | (ys , y , refl) = ys
+init {n = zero}  (x ∷ xs) = []
+init {n = suc n} (x ∷ xs) = x ∷ init xs
 
 last : Vec A (1 + n) → A
-last xs with initLast xs
-... | (ys , y , refl) = y
+last {n = zero}  (x ∷ xs) = x
```

Comment:
I still think in this clause that we should have
```agda
last {n = zero} (x ∷ []) = x
```
It's more (perhaps needlessly?) strict, but documents the situation enforced by the index being 1+0...

Reply:
Alternatively, use an underscore...

Reply:
Sure, I changed `xs` in the first clause to `_`.